package.json

```diff
@@ -132,12 +135,23 @@
 		},
 		"verbose": true
 	},
+	"lint-staged": {
+		"*.js": [
+			"prettier --write",
+			"eslint --fix",
```

Comment:
There's no use to run `eslint --fix` command with the current `.eslintrc.json` file.

_—Why?_

❌  Basically, this is what's happening here

- Prettier fixes and rewrites the files
- ESLint again fixes them with many overlapping formatting rules (potentially doing again what it was doing without prettier)

✅  What should happen here?

- Prettier fixes and rewrites the files
- ESLint fixes what Prettier can't
- ESLint leaves formatting of all the overlapping rules to prettier using something like [`prettier-eslint`](https://github.com/prettier/prettier-eslint/) but that's not it, we also need [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) to do that and to format only the conflicting rules, we need to fork `eslint-config-prettier` or use before and add more rules custom rules after that where Prettier fails WP coding standards.

This is a draft of what I think and where I am stuck while trying to implement Prettier in [`create-guten-block`](https://github.com/ahmadawais/create-guten-block).

Reply:
And what is wrong with that? It's just CPU cycles 😉 

It's pretty quick as it is only _pretty printing_ and linting _staged files_

I honestly don't think `prettier-eslint` is all that helpful, it requires a _formatter_ that can _pretty print_ in the Editor or IDE, a _formatter_ plugin is not available for some of the common IDEs and Editors out there, the configuration of each is also different, this adds another burden on developers to configure _another_ tool (See also this [@JJJ Tweet](https://twitter.com/JJJ/status/955249428394381312) from earlier today)

The `eslint-plugin-prettier` from what I gather does what it says in the description of the plugin _"Runs Prettier as an ESLint rule and reports differences as individual ESLint issues."_, now users see even more ESLint warnings coming from Prettier via the ESLint command. 

If we can hide as many errors from developers we can the better for all concerned. 

So, to that end, I don't see an issue having Prettier then ESLint fixing and writing the files, It's pretty quick as it is only performing this on _staged files_

There is no extra tooling burden added to developers with this approach, and it's probably not ideal but it's just robots processing code, actually, just a few CPU cycles 😏

Reply:
It would be a good idea to add both eslint-plugin-prettier and eslint-config-prettier and see if it integrates well. That way you would have to run ‘eslint —fix’ only as a precommit hook. See this article: https://medium.com/@netczuk/your-last-eslint-config-9e35bace2f99.

Reply:
The problem with `eslint-config-prettier` is it turns off rules we want to keep, e.g [space-in-parens](https://github.com/prettier/eslint-config-prettier/blob/master/index.js#L76).

The plugin `eslint-plugin-prettier` uses Prettier to _pretty print_ so why not just let Prettier do this and not add another dependency? I've honestly struggled to see any benefit at all of this plugin to be honest, all it does it shows extra ESLint errors for issues that Prettier is required to fix, to me, it seems to just add more noise /shrug

Reply:
>And what is wrong with that? It's just CPU cycles 😉

I think it's more than just that. When an IDE picks up conflicting tools then you end up in a loop of one tool fixing an issue and creating that issue for another tool and vice versa.

>The eslint-plugin-prettier from what I gather does what it says in the description of the plugin "Runs Prettier as an ESLint rule and reports differences as individual ESLint issues.", 

No actually, it does not. It does that in a combination with `eslint-config-prettier` without that, it's a loop of one tool fixing and breaking things for another.

>There is no extra tooling burden added to developers with this approach, and it's probably not ideal but it's just robots processing code, actually, just a few CPU cycles 

Again the problem here is the same. Also, I kind of dislike linting + formatting on staged files. That way something wrong can end up in the code base and all you have to say about that is I didn't write it that way — but the tools ended up linting it all at the end.

Also since the codebase is not just getting started — this can pose more problems. Having it run progressively via editors is my preferred way of doing things. But that's just me.

Right now what worries me here is the one tool's fixes are another tool's problems!

@ntwb @gziolo I am actually trying to make this doable in a single package. I'd love to do it with you folks and then we can add it here as a single dependency.

Reply:
My "It's just CPU cycles" has nothing to do with editors or IDEs, its run by Git as a commit hook

FYI: This PR isn't just about Gutenberg, it's also about the feasibility of adding it to WordPress core, for the most part this will be in part achievable by us sharing this tooling via the `@wordpress/packages` library that @gziolo has already mentioned, our goal here is to have it available to share with plugin and theme devs, so once we've got it sorted, you'll be able to also use it in your package.

p.s. See my other comment on why eslint plugin/config-prettier are not suitable for us